app/server.py

```diff
@@ -19,108 +20,269 @@
+	or all wishlists when no id is specified.
+	Example: curl http://127.0.0.1:5000/wishlists/1
+	"""
+	try:
+		wl = Wishlist.find_or_404(wishlist_id)
+		return make_response(jsonify(wl.serialize_wishlist()), status.HTTP_200_OK)
+	except WishlistException:
+		return make_response(jsonify(message='Cannot retrieve wishlist with id %s' % wishlist_id), status.HTTP_404_NOT_FOUND)
 
 
-@app.route('/wishlists/<int:wishlist_id>/items',methods=['POST'])
-def add_item_to_wishlist(wishlist_id):
-    """
-    The route for adding new items to the wishlist. This method can also be checked using CURL.
-    Pre-requisite: Create a wishlist to add an item.
-    Example: curl -i -H 'Content-Type: application/json' -X POST -d '{"id":"i123","description":"Awesome product!"}' http://127.0.0.1:5000/wishlists/1/items
-    curl -i -H 'Content-Type: application/json' -X POST -d '{"id":"i12","description":"Apple product!"}' http://127.0.0.1:5000/wishlists/1/items
-    """
-    data = request.get_json()
-    if is_valid(data,'item'):
-        try:
-        	wl = Wishlist.find_or_404(wishlist_id)
-        	wl.deserialize_wishlist_items(data)
-        	wl.save_item()
-        	message = wl.serialize_wishlist()    
-		return make_response(jsonify(message), status.HTTP_201_CREATED, {'Location': wl.self_url()})
-        except WishlistException:
-            return make_response(jsonify(message='Cannot add a new item %s' % request.json['id']), status.HTTP_400_BAD_REQUEST)
-    else:
-        message = { 'error' : 'Wishlist %s was not found' % wishlist_id }
-        return make_response(jsonify(message), status.HTTP_404_NOT_FOUND)
+@app.route('/wishlists/<int:wishlist_id>/items', methods=['GET'])
+def item(wishlist_id):
+	"""
+	The route for getting all items associated with a wishlist
+	or making a new item for a wishlist via a POST.
+	Example: curl http://127.0.0.1:5000/wishlists/1/items
+	"""
+	try:
+		wl = Wishlist.find_or_404(wishlist_id)
+		items = wl.all_items()
+		return make_response(jsonify(items), status.HTTP_200_OK)
+	except WishlistException:
+		return make_response(jsonify(message='Cannot retrieve wishlist with id %s' % wishlist_id), status.HTTP_404_NOT_FOUND)
+
+
+@app.route('/wishlists/<int:wishlist_id>/items/<string:item_id>', methods=['GET'])
+def read_wishlist_item(wishlist_id, item_id):
+	"""
+	The route for retrieving a specific item in a wishlist.
+	Example: curl http://127.0.0.1:5000/wishlists/1/items/i123
+	"""
+
+	try:
+		wl = Wishlist.find_or_404(wishlist_id)
+		item = wl.find_item(item_id)
+		return make_response(jsonify(item), status.HTTP_200_OK)
+	except ItemException:
+		return make_response(jsonify(message='Item with id %s could not be found' % item_id), status.HTTP_404_NOT_FOUND)
+	except WishlistException:
+		return make_response(jsonify(message='Wishlist with id %d could not be found' % wishlist_id), status.HTTP_404_NOT_FOUND)
+
+
+@app.route('/wishlists/<int:id>', methods=['PUT'])
+def update_wishlist(id):
```

Comment:
When update a wishlist, maybe just update its name or user_id, the items in this wishlist gone.